PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,584 @@
+        }
+    }
+
+    Context '\n' {
+
+        It 'matches new lines' {
+            <#
+                The \n symbol is the first regex symbol you've encountered that isn't just a single
+                character. Normally if you saw a lowercase "n" in a regex, it would just mean
+                "literally the letter n". When it's preceded by a backslash, however, that "n" takes
+                on a special meaning. In this case, \n matches new lines. Actually, it matches the
+                encoded character that indicates a new line.
+    
+                There are many more regex symbols that are single letters preceded by a backslash.
+                In fact, the backslash is probably the single most important character in all of
+                regex.
+            #>
+    
+            $multiLine = @"
+They might look
+similar and look
+soft, but you should
+never confuse a
+wild cougar for
+a domestic cat
+"@
+            $catPattern = '____\n'
+            $multiLine -match $catPattern | Should -BeTrue
+        }
+    }
+
+    Context '\d and \D' {
+
+        It 'matches digits' {
+            <#
+                The \d symbol matches digits (the numbers 0 - 9 inclusive), while the symbol \D
+                matches anything OTHER THAN a digit. Yes, regular expressions are case sensitive.
+                This "lower case for affirmative, upper case for negative" pattern is common among
+                regex symbols as you'll discover as you carry on.
+            #>
+    
+            '__' -match '\d' | Should -BeTrue
+        }
+
+        It 'can require more than one occurence in a row' {
+            '__' -match '\d\d\d' | Should -BeTrue
+        }
+
+        It 'matches things that are NOT digits' {
+            '__' -match '\D' | Should -BeTrue
+        }
+    }
+
+    Context '\w and \W' {
+
+        It 'matches (non-)word characters' {
+            <#
+                \w matches word characters which are letters, numbers, and underscores. The \W
+                symbol matches anything else.
+            #>
+            '***' -match '____' | Should -BeTrue
+        }
+
+        It 'sees no funky characters in here'
+            'warmth' -match '____' | Should -BeTrue
+        }
+    }
+
+    Context '^ (caret) and $ (dollar sign)' {
+
+        It 'matches the start of lines' {   
+            $startAndEnd = '____'
+            $startAndEnd -match '^a' | Should -BeTrue
+        }
+
+        It 'matches the end of lines' {
+            $startAndEnd = '____'
+            $startAndEnd -match 'z$' | Should -BeTrue
+        }
+    }
+
+    Context '\s and \S' {
+
+        It 'matches (non-) whitespace characters' {
+            <#
+                \s matches whitespace characters (space, tab, etc.) while \S matches anything other
+                than whitespace
+            #>
+    
+            # Enter either \s or \S - which do you think will work?
+            'Room to grow' -match '____' | Should -BeTrue
```

Comment:
I kind of want them to think about the fact that they have to type the backslash though. I feel like that would be too much spoon-feeding. I could be convinced otherwise, though.

Reply:
Backslash I can take or leave but I do think we should be using a shorter blank :)